lib/plugins/aws/package/compile/events/cloudFront.js

```diff
@@ -444,16 +444,28 @@ class AwsCompileCloudFrontEvents {
               Object.assign(behavior, event.cloudFront.behavior);
             }
 
+            if (
+              (event.cloudFront.cachePolicy ||
+                (event.cloudFront.behavior && event.cloudFront.behavior.CachePolicyId)) &&
+              !shouldAssignCachePolicy
+            ) {
+              throw new ServerlessError(
+                'Both cachePolicy and deprecated fields ForwardedValues, MinTTL, MaxTTL' +
+                  'and DefaultTTL found in function ${functionObj.name} configuration.' +
+                  'Specifying a cachePolicy override those deprecated parameters.' +
+                  'Please remove one of the cache behavior definition.',
+                'CACHE_POLICY_ID_AND_DEPRECATED_FIELDS_USED'
+              );
+            }
+
+            if (event.cloudFront.behavior && event.cloudFront.behavior.CachePolicyId) {
+              Object.assign(behavior, {
+                CachePolicyId: event.cloudFront.behavior.CachePolicyId,
+              });
+              shouldAssignCachePolicy = false;
+            }
+
             if (event.cloudFront.cachePolicy) {
```

Comment:
If I'm not mistaken, in the current logic, if the `cachePolicy.name` is specified, it will also override `behavior.CachePolicyId` setting, correct? I think it would be good to document it as well

Reply:
@pgrzesik Thanks for the quick reply 👍 Yeah, that's correct. Let me improve PR and let you know :)

Reply:
@pgrzesik I just pushed a change. I considered adding some more tests for this `name` property case. In the end, I thought that it can be redundant to add more tests for it as the `name` case would have almost the same call stack trace as the `id` case. Could you let me know what do you think? :bow:

And not sure I updated the document in a well-aligned way to the community convention. Please feel free to let me know how I can polish the phrases :bow:

Reply:
I think it might be nice to cover it with a single test where `cachePolicy.name` takes precedence over `behavior.CachePolicyId`.

Reply:
That will be more solid. Thanks for the guidance 🙇 I will update the PR and let you know.